src/utils/FourWay.js

```diff
@@ -444,6 +444,15 @@ class FourWay {
       } else {
         let readbackSettings = null;
         if(esc.isSiLabs) {
+          const mcu = esc.settings.MCU;
+          if (mcu && mcu.startsWith('#BLHELI$EFM8')) {
+            const CODE_LOCK_BYTE_OFFSET = mcu.endsWith('B21#') ? 0xFBFE : 0x1FFE;
+            const codeLockByte = (await this.read(CODE_LOCK_BYTE_OFFSET, 1)).params[0];
+            if (codeLockByte !== 0xFF) {
+              throw new Error(`ESC is locked (${codeLockByte})`);
```

Comment:
The remaining code in this function throws errors like this, but I'm not sure if it is properly logged this way?

Reply:
No. Throwing will not automatically log. Whoever calls this method needs to handle catching the errors. In this case, no one is catching, so it will fail hard.

This is still on my todo - Errors are thrown and the error messages are always in english, which is not so great for internationalization. My plan was to throw custom errors and then log according to error type, only passing arguments to the thrown error instead of text:

```javascript
throw new EscLockedError(codeLockByte);
```

and then use the instance of the Error as the base for the translation.

In some cases I catch the Error 'locally' and log a matching error message...

Reply:
Okay, I will leave it as it is for now.